apps/elixir_ls_debugger/lib/debugger/server.ex

```diff
@@ -284,9 +284,12 @@ defmodule ElixirLS.Debugger.Server do
     {%{"variables" => vars_json}, state}
   end
 
-  defp handle_request(request(_, "evaluate"), state) do
-    msg = "(Debugger) Expression evaluation in Elixir debugger is not supported (yet)."
-    {%{"result" => msg, "variablesReference" => 0}, state}
+  defp handle_request(request(_cmd, "evaluate", %{"expression" => expr} = _args), state) do
+    {term, _bindings} = Code.eval_string(expr)
```

Comment:
Let's wrap it in a task. Otherwise a throw or an exit will bring down the debugger. If a task won't work for some reason at least we need to also catch throws beside rescuing exceptions

Reply:
Do you mean mix task and launching it with `Mix.Task.run("task_name", [code_str])`?

Reply:
I meant using something like error kernel pattern
```
owner = self()
{:ok, pid} = Task.start_nolink(fn ->
  try do
    {term, _bindings} = Code.eval_string(expr)
    send(owner, {:ok, term})
  catch
    error -> send(owner, {:throw, error})
    kind, error -> send(owner, {kind, error})
  end
end)
Process.monitor(pid)

receive do
  {:ok, term} -> do_sth_with(term)
  {kind, error} -> report_error(kind, error)
  {:DOWN, _, :process, ^pid, reason} -> report_error(reason)
after
  timeout ->
    Process.exit(pid, :normal)
    report_timeout()
end
```

Reply:
Okay, got it! Thank you for feedback :)

I'll improve soon